File: csharp/ql/test/query-tests/Bad Practices/Declarations/NoConstantsOnly/NoConstantsOnly.cs
using System;

class NoConstantsOnly
{
    abstract class MathConstants
    {
        public const double Pi = 3.14; // BAD
    }

    class Circle : MathConstants
    {
        private double radius;

        public double Area() => Math.Pow(radius, 2) * Pi;
    }
}

File: java/ql/src/Violations of Best Practice/Declarations/NoConstantsOnly.java
public class NoConstantsOnly {
	static interface MathConstants
	{
	    public static final Double Pi = 3.14;
	}

	static class Circle implements MathConstants
	{
	    public double radius;
	    public double area()
	    {
	        return Math.pow(radius, 2) * Pi;
	    }
	}
}